tasks/test_results_finisher.py

```diff
@@ -296,121 +283,43 @@ def process_impl_within_lock(
-                )
-
-            metrics.incr(
-                "test_results.finisher.flaky_test_detection",
-                tags={"status": success, "reason": reason},
-            )
-
         return {
             "notify_attempted": True,
             "notify_succeeded": success,
             QUEUE_NOTIFY_KEY: False,
         }
 
-    def run_flaky_test_detection(
+    def get_flaky_tests(
         self,
-        db_session,
-        repoid,
-        notifier: TestResultsNotifier,
-        payload: TestResultsNotificationPayload,
-        checkpoints=None,
+        db_session: Session,
+        commit_yaml: UserYaml,
+        repoid: int,
+        commit: Commit,
+        failures: list[TestResultsNotificationFailure],
     ):
-        ignore_predefined = read_yaml_field(
-            "test_analytics", "ignore_predefined", _else=False
-        )
-
-        user_normalization_regex = read_yaml_field(
-            "test_analytics", "normalization_regex", _else=dict()
-        )
-
-        failure_normalizer = FailureNormalizer(
-            user_normalization_regex, ignore_predefined
-        )
-
-        default_branch_failure_detector = DefaultBranchFailureDetector(
-            db_session, repoid, "main"
-        )
-        unrelated_matches_detector = UnrelatedMatchesDetector(failure_normalizer)
-        diff_outcome_detector = DiffOutcomeDetector()
-
-        flake_detection_engine = FlakeDetectionEngine(
-            db_session,
-            repoid,
-            [
-                default_branch_failure_detector,
-                unrelated_matches_detector,
-                diff_outcome_detector,
-            ],
-        )
-
-        log.info(
-            "Starting flake detection",
-            extra=dict(
-                repoid=repoid,
-                parent_task=self.request.parent_id,
-            ),
-        )
-        current_state_of_repo_flakes = flake_detection_engine.detect_flakes()
-
-        for test_id, symptoms in current_state_of_repo_flakes.items():
-            log.info(
-                "Discovered flaky test",
-                extra=dict(
-                    repoid=repoid,
-                    parent_task=self.request.parent_id,
-                    test_id=test_id,
-                    symptoms=list(symptoms),
-                ),
-            )
-            payload.flaky_tests[test_id] = TestResultsNotificationFlake(
-                list(symptoms),
-                True,
+        if FLAKY_TEST_DETECTION.check_value(
+            identifier=repoid, default=False
+        ) and read_yaml_field(
+            commit_yaml, ("test_analytics", "flake_detection"), False
+        ):
+            flaky_tests = set()
+            repo_flakes = (
+                db_session.query(Flake)
+                .filter(  # type:ignore
+                    Flake.repoid == repoid,
+                    Flake.end_date.is_(None),
+                )
+                .all()
```

Comment:
I agree that we could limit and offset this query to reduce memory usage but maybe it'd be better to filter on test ids in the query like i mentioned in my comment below, and also only select the test id from the query which is all we're looking for here. Is there another reason other than memory usage that we would limit this query?

Reply:
>  only select the test id from the query which is all we're looking for here.

I'm good with providing a id list. We will probably also want to ensure that the list isn't too long (i.e. we can cap this on the application side). It's _probably_ fine to query all the tests for now. However, we will want to keep an eye on the performance of this query (correlated with number of failed tests it's trying to retrieve).

> Is there another reason other than memory usage that we would limit this query?

Yes. Having a bound on the number of items returned will also make the processing of the results more predictable. For example, we don't have to worry (as much) about long running tasks because it's going to only ever process i.e. 30 items.